ui-core/src/components/inputs/InputWithSuggestions.tsx

```diff
@@ -0,0 +1,107 @@
+import React, {
+  useState,
+  useRef,
+  useEffect,
+  ChangeEventHandler,
+  KeyboardEventHandler,
+  ReactNode,
+} from 'react';
+import cx from 'classnames';
+import Input, { InputProps } from './Input';
+
+interface InputWithSuggestionsProps<T> extends Omit<InputProps, 'onChange' | 'value'> {
+  suggestions: Array<T>;
+  onChange: (value: T) => void;
+  getSuggestionLabel: (option: T) => string;
+  getSuggestionKey: (value: T) => string;
+  customLabel?: ReactNode;
+}
+
+function InputWithSuggestions<T>({
+  suggestions,
+  onChange,
+  getSuggestionLabel,
+  getSuggestionKey,
+  customLabel,
+  ...inputProps
+}: InputWithSuggestionsProps<T>) {
+  const [filteredSuggestions, setFilteredSuggestions] = useState<T[]>([]);
+  const [activeSuggestion, setActiveSuggestion] = useState<number>(-1);
+  const [showSuggestions, setShowSuggestions] = useState<boolean>(false);
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const handleInputChange: ChangeEventHandler<HTMLInputElement> = (e) => {
+    const userInput = e.currentTarget.value;
+
+    if (userInput === '') {
+      setFilteredSuggestions([]);
+      setShowSuggestions(false);
+      return;
+    }
+
+    const filtered = suggestions.filter((suggestion) =>
+      getSuggestionLabel(suggestion).toLowerCase().startsWith(userInput.toLowerCase())
```

Comment:
In other search input in osrd, we can search an item even if it doesn't start by the input typed (ex : I type "antes" and I see in suggestions "Mantes". I know it's not in sketch requirements but maybe we should ask @thibautsailly if it's needed ?